Sources/TuistScaffold/TemplatesDirectoryLocator.swift

```diff
@@ -15,7 +15,7 @@ public final class TemplatesDirectoryLocator: TemplatesDirectoryLocating {
     private let rootDirectoryLocator: RootDirectoryLocating
 
     /// Default constructor.
-    public init(rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator.shared) {
```

Comment:
What about removing `.shared` from `RootDirectoryLocator`. In hindsight, I think it was a good idea introducing that there because that class holds state, and that might introduce side effects to both the code and the tests.